### PR TITLE
Support local k8s-netperf execution

### PR DESCRIPTION
**Pull request by sjug**
## Type of change

- [x] Refactor
- [x] New feature
- [x] Bug fix
- [x] Optimization
- [x] Documentation Update

## Description

Needed to add the rest of the vars that are hardcoded/missing but thought it may be handy to have a way to exclude the var from the command.
- Refactor parameter generation
- Add new variables & sort alphabetically
- Remove `arch` command to get system architecture
- Minor checks and style changes

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
MicroShift 4.14.5

- Please provide detailed steps to perform tests related to this code change.
Try to run `run.sh`, changing environment variables optional.

### CODE REVIEW

File: workloads/network-perf-v2/env.sh [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 
 # k8s-netperf version
+export ALL_SCENARIOS=${ALL_SCENARIOS:-true}
+export DEBUG=${DEBUG:-true}
+export METRICS=${METRICS:-true}
+export NETPERF_FILENAME=${NETPERF_FILENAME:-k8s-netperf}
 export NETPERF_VERSION=${VERSION:-v0.1.18}
 export OS=${OS:-Linux}
-export ARCH=$(arch)
+export ARCH=$(uname -m)
```
Comment (jtaleric) [inline workloads/network-perf-v2/env.sh:14]: not sure the container images in Prow have `uname` -- I think we used to have `uname` but ran into issues in Prow? I might be misremembering though.

File: workloads/network-perf-v2/run.sh [hunk 2]
```diff
@@ -4,7 +4,10 @@ set -e
 source ./env.sh
 source ../../utils/common.sh
 
-curl -sS -L $NETPERF_URL | tar -xz
+# Download and extract k8s-netperf if we haven't already
+if [ ! -f "${NETPERF_FILENAME}" ]; then
```
Comment (jtaleric) [inline workloads/network-perf-v2/run.sh:8]: hm, could this cause issues if people are flipping versions of k8s-netperf?

**Review by jtaleric — COMMENTED** [review]
Thanks for this! Lets get a rebase.

File: workloads/network-perf-v2/env.sh (hunk 1 above)
Comment (sjug) [reply workloads/network-perf-v2/env.sh:14]: Pretty sure they're both in `coreutils-single` in RHEL. `arch` is more RHEL specific where `uname` is everywhere.

File: workloads/network-perf-v2/run.sh (hunk 2 above)
Comment (sjug) [reply workloads/network-perf-v2/run.sh:8]: Gimme a `k8s-netperf --version` flag to work with?

File: workloads/network-perf-v2/env.sh (hunk 1 above)
Comment (jtaleric) [reply workloads/network-perf-v2/env.sh:14]: 😿 I think they don't use RHEL container images... 

Looking at the output from one of the runs in prow.
```
+ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
+ oc config view
```

File: workloads/network-perf-v2/run.sh (hunk 2 above)
Comment (jtaleric) [reply workloads/network-perf-v2/run.sh:8]: Fair ;)

File: workloads/network-perf-v2/env.sh (hunk 1 above)
Comment (sjug) [reply workloads/network-perf-v2/env.sh:14]: Not using RHEL container images, yikes.

```
$ podman run -it debian:11
root@21182cf1c235:/# uname -m
x86_64
root@21182cf1c235:/# which uname
/bin/uname
```

File: workloads/network-perf-v2/run.sh (hunk 2 above)
Comment (jtaleric) [reply workloads/network-perf-v2/run.sh:8]: Done.

**Comment by jtaleric** [comment]
@sjug we can bump the release to v0.1.20 and that will have `--version` and we can get this merged in.

**Comment by sjug** [comment]
Rebased and updated.

File: workloads/network-perf-v2/run.sh [hunk 3]
```diff
@@ -23,13 +39,38 @@ log "###############################################"
...
+    cmd+=" --$flag_name $flag_value"
+  fi
+}
+
+# Add flags based on conditions
+add_flag "all" "${ALL_SCENARIOS}"
+add_flag "clean" "${CLEAN_UP}"
+add_flag "config" "${WORKLOAD}"
+add_flag "debug" "${DEBUG}"
+add_flag "local" "${LOCAL}"
```
Comment (jtaleric) [inline workloads/network-perf-v2/run.sh:60]: if we set local, lets disable all.

File: workloads/network-perf-v2/run.sh [hunk 4]
```diff
@@ -23,13 +39,38 @@ log "###############################################"
...
+
+# Initialize the command var
+cmd="timeout ${TEST_TIMEOUT} ./k8s-netperf"
+
+# Function to add flags conditionally
+add_flag() {
+  local flag_name="$1"
+  local flag_value="$2"
+  if [ -n "$flag_value" ]; then
+    cmd+=" --$flag_name $flag_value"
```
Comment (jtaleric) [inline workloads/network-perf-v2/run.sh:51]: Due to the bool issue, I think we need to add `--$flag_name=$flag_value`

File: workloads/network-perf-v2/run.sh (hunk 3 above)
Comment (sjug) [reply workloads/network-perf-v2/run.sh:60]: Added

**Review by jtaleric — APPROVED** [review]
Nice! Thanks! LGTM

**Review by mukrishn — APPROVED** [review]
lgtm

**Pull request merged**